enhancements/installer/assisted-installer-bare-metal-validations.md

```diff
@@ -0,0 +1,228 @@
+inputs needed for the assisted installation workflow.
+
+#### Story 2
+
+As a developer of the OpenShift installer, I want to be able to add
+validation rules for bare metal clusters correctly based on whether
+the cluster is being built using the installer-provisioned
+infrastructure workflow or the assisted installer workflow so I can
+apply only the rules necessary for each case.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The install config inputs are currently validated as part of
+generating the `InstallConfig` asset in
+[InstallConfig.finish()](https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/installconfig.go#L141),
+which invokes some logic that applies to all platforms in
+[ValidateInstallConfig()](https://github.com/openshift/installer/blob/master/pkg/types/validation/installconfig.go#L40)
+before running the platform-specific logic from
+[validatePlatform()](https://github.com/openshift/installer/blob/master/pkg/types/validation/installconfig.go#L341). For
+the `baremetal` platform that results in calling
+[ValidatePlatform()](https://github.com/openshift/installer/blob/master/pkg/types/baremetal/validation/platform.go#L218).
+
+The
+[PlatformProvisionCheck](https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/platformprovisioncheck.go)
+and
+[PlatformCredsCheck](https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/platformcredscheck.go)
+assets are both dependencies for the `Cluster` asset, which means they
+are evaluated later in the installation process. Both already have
+platform-specific logic, although not for the `baremetal` platform.
+
+The assisted installer does not need the `bootstrapProvisioningIP`,
+`provisioningNetworkCIDR`, `provisioningDHCPRange`, or
+`provisioningNetworkInterface` fields. The values in those fields are
+used to configure the provisioning tools used for the
+installer-provisioned infrastructure workflow, so those validation
+checks could move to a new function in the
+`pkg/types/baremetal/validation` package that can be invoked from
+[PlatformProvisionCheck.Generate()](https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/platformprovisioncheck.go#L37).
+
+There are two validation rules for the `clusterOSImage` and
+`bootstrapOSImage` fields, to ensure the value is a valid URL and to
+ensure the URL exists. Those checks can be split between the two
+phases, so that the URL format validation is done in
+`ValidateInstallConfig()` and the existence check is moved to
+`PlatformProvisionCheck`.
+
+The assisted installer does not need the `libvirtURI` or host BMC
+credentials, so the checks for those values could move to a new public
+function in the `pkg/types/baremetal/validation` package that can be
+invoked from
+[PlatformCredsCheck.Generate()](https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/platformcredscheck.go#L60)
```

Comment:
Should we link to the upstream work to allow BMC credentials to be empty?

https://github.com/metal3-io/baremetal-operator/pull/563

Reply:
added